### DISCUSSION TITLE
Route should only match dynamic params if a slash is immediately followed by a colon

### DISCUSSION THREAD

**Discussion by Murderlon** [mswjs/msw#1293] [Ideas]
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 14 or higher

### Browsers

Firefox

### Reproduction repository

https://codesandbox.io/s/eloquent-panna-n2duee?file=/src/mocks/handlers.js

### Reproduction steps

See the codesandbox.

Routes should only be dynamic params if a slash is immediately followed by a colon (`/:foo/bar`, not `/foo:bar/baz`)

### Current behavior

In the codesandbox I request the **second** handler but it resolves with weird params to the **first**.

### Expected behavior

Only have dynamic params if a slash is immediately followed by a colon

**Comment by Murderlon** [comment]
Want to say that I love this library so far and I'm willing to contribute a fix if we agree it's a problem.

**Comment by kettanaito** [comment]
Hey, @Murderlon. Thanks for raising this. 

MSW uses a [path-to-regexp](https://www.npmjs.com/package/path-to-regexp) package, which is also used by Express and acts as a foundation for a new `URLPattern` feature that standardizes URL path parameters. This means that the behavior you're experiencing is how `path-to-regexp` works, there isn't much we can/should do about it. 

```js
var { match } = require("path-to-regexp")

const fn = match('/foo:bar')
fn('/foosome') // params: { bar: "some" }
```

> [Try it on RunKit](https://npm.runkit.com/path-to-regexp)

That being said, I find this behavior to be correct. Although they are often referred to as "path" parameters, these dynamic segments can be present anywhere in the URL string. For instance, you can parametrize your hostname in a similar fashion:

```
https://www.:environment.service.com/path
```

So I can't agree that path parameters should apply only when preceded by a slash.

**Reply by Murderlon** [reply]
Makes sense, thanks for explaining.
